src/components/ObjectInput.tsx

```diff
@@ -25,56 +29,58 @@ export const ObjectInput: React.FunctionComponent<IFormtronControl> = ({
   const noConflict = (key: any) => !(key in value);
 
   return (
-    <Box as="fieldset" position="relative">
-      <legend>{schema.title}</legend>
-      {easyObject.items.map((entry, index) => {
-        const [key, val] = entry;
-        const _selection = selection === '' || selection === '.' ? `${index}` : `${selection}.${index}`;
-        return (
-          <Flex key={`${index}-${easyObject.items.length}`}>
-            <Flex flexDirection="column">
-              <Button type="button" title="Insert item" onClick={() => onChange(easyObject.insert(index))}>
-                <Text color="green">+</Text>
-              </Button>
-              <Button type="button" title="Delete item" onClick={() => onChange(easyObject.remove(index))}>
-                <Text color="red">x</Text>
-              </Button>
+    <Messages path={path}>
+      <FieldSet position="relative" invalid={variant === 'invalid'} legend={schema.title}>
+        {easyObject.items.map((entry, index) => {
+          const [key, val] = entry;
+          return (
+            <Flex key={`${index}-${easyObject.items.length}`}>
+              <Flex flexDirection="column">
+                <Button type="button" title="Insert item" onClick={() => onChange(easyObject.insert(index))}>
+                  <Text color="green">+</Text>
+                </Button>
+                <Button type="button" title="Delete item" onClick={() => onChange(easyObject.remove(index))}>
+                  <Text color="red">x</Text>
+                </Button>
+              </Flex>
+              <DraftValue
+                value={key}
+                onChange={_key => noConflict(_key) && onChange(easyObject.updateKey(index, _key))}
```

Comment:
Consider using `useCallback` for better readability.

Reply:
I thought `useCallback` was for memoizing things?

Reply:
What I meant is to have sth like
```
const handleChange = (_key) => {
  if (noConflict(_key)) {
    onChange(easyObject.updateKey(index, _key));
  }
};
```

not opinionated on that, though IMHO it would be more readable